server.go

```diff
@@ -67,7 +68,9 @@ func (svr *Server) getHandle(handle string) (*os.File, bool) {
 
 type serverRespondablePacket interface {
 	encoding.BinaryUnmarshaler
+	id() uint32
 	respond(svr *Server) error
+	writing() bool
```

Comment:
Yeah, my intent was to make the zero-value the "read-only" one, but I suppose I can invert it.